src/pages/NIHicWebform.js

```diff
@@ -0,0 +1,2502 @@
+      return prev;
+    });
+  };
+
+  onOntologiesChange = (data) => {
+    this.setState(prev => {
+      prev.formData.ontologies = data || [];
+      return prev;
+    });
+  };
+
+  setOther = () => {
+    this.setState(prev => {
+      prev.formData.generalUse = false;
+      prev.formData.hmb = false;
+      prev.formData.diseases = false;
+      prev.formData.other = true;
+      prev.formData.ontologies = [];
+      return prev;
+    });
+  };
+
+  setOtherText = (e) => {
+    const value = e.target.value;
+    this.setState(prev => {
+      prev.formData.other = true;
+      prev.formData.otherText = value;
+      return prev;
+    });
+  };
+
+  back = () => {
+    this.props.history.goBack();
+  };
+
+  dacOptions = () => {
+    return this.state.dacList.map(function(item) {
+      return {
+        key: item.dacId,
+        value: item.dacId,
+        label: item.name,
+        item: item
+      };
+    });
+  };
+
+  onDacChange = (option) => {
+    this.setState(prev => {
+      if (fp.isNil(option)) {
+        prev.selectedDac = {};
+        prev.datasetData['dac'] = '';
+      } else {
+        prev.selectedDac = option.item;
+        prev.datasetData['dac'] = option.label;
+        prev.disableOkBtn = false;
+        prev.problemSavingRequest = false;
+        prev.problemLoadingUpdateDataset = false;
+      }
+      return prev;
+    });
+  };
+
+  formatOntologyItems = (ontologies) => {
+    return ontologies.map((ontology) => {
+      return {
+        id: ontology.id || ontology.item.id,
+        key: ontology.id || ontology.item.id,
+        value: ontology.id || ontology.item.id,
+        label: ontology.label || ontology.item.label,
+        definition: ontology.definition || ontology.item.definition,
+        item: ontology || ontology.item
+      };
+    });
+  };
+
+  render() {
+
+    const controlLabelStyle = {
+      fontWeight: 500,
+      marginBottom: 0
+    };
+
+    const {
+      hmb = false,
+      poa = false,
+      diseases = false,
+      other = false,
+      otherText = '',
+      genetic = false,
+      forProfit = false,
+      publication = false,
+      collaboration = false,
+      ethics = false,
+      geographic = false,
+      moratorium = false,
+      methods = false,
+      generalUse = false,
+    } = this.state.formData;
+    const ontologies = this.formatOntologyItems(this.state.formData.ontologies);
+    const { publicAccess = false } = this.state.datasetData;
+    const { npoa } = !poa;
+    const { problemSavingRequest, problemLoadingUpdateDataset, showValidationMessages, submissionSuccess } = this.state;
+    const isTypeOfResearchInvalid = false;
+    const isUpdateDataset = (!fp.isEmpty(this.state.updateDataset));
+    const profileUnsubmitted = span({}, [
+      'Please make sure ',
+      h(Link, { to: '/profile', className: 'hover-color' }, ['Your Profile']),
+      ' is updated, as it will be linked to your dataset for future correspondence'
+    ]);
```

Comment:
Since this is a demo page, there won't be a logged in user. This logic and usage should be removed.